### ISSUE TITLE
Getting HTTP Error 404 message not found since adding a token

### ISSUE THREAD

**Issue by crovaxon** [phyzical/docker.versions#12]
**Describe the bug**
Following the suggestion to add a Github token to avoid being API ratelimited I generated a token for myself and added it to the plugin.
Unfortunately since then it produces only error messages in the changelog.

**To Reproduce**

1. Create token (tried both classic token or new fine-grained tokens), did not give any additional permissions in the token
2. Add token to plugin, confirms that the token is successfully set
3. Check changelog on any docker container with an update available
4. Get 404 HTTP errors in changelog

**Expected behavior**
The usual (and incredibly useful) functionality of generating a changelog between docker image versions.

**Screenshots**
![image](https://github.com/user-attachments/assets/b3fc1563-ebfe-4a5a-bbdc-c3f983641c0c)
No matter what docker image with an update I check, they all error out the same way.

**Unraid version**
6.12.10

**Any logs**

* /var/log/phplog - for php error logs
* right click and inspect check the console for js errors

Browser console does not show any errors while operating the changelog feature.
I don't have a /var/log/phplog.

**Additional context**

Any way to also unset a token in the plugin? Being able to reset the saved token could be useful perhaps.

**Comment by phyzical** [comment]
@crovaxon you should be able to "unset" by deleting the save file `rm /boot/config/docker.versions/github_token.txt`

Ill look at adding a button to do this for future people but good idea.

Before you do though, lets see if we can workout whats happened. can you check that it has saved as you expected at `/boot/config/docker.versions/github_token.txt` ?

So `cat /boot/config/docker.versions/github_token.txt` in unraid

LMK if the value looks right (obviously dont paste it), make sure theres no whitespace or something weird.

Ill try setting it on my install again incase something has changed that i overlooked

**Comment by phyzical** [comment]
hmm infact if it was an auth issue you would be seeing the following (401 not 404)
<img width="829" alt="image" src="https://github.com/user-attachments/assets/457d02a2-dfec-4fcc-8398-e92cfbb84b05">

**Comment by phyzical** [comment]
So i think your token is correct,

the repo its pulled looks like one that needs the bridging, so all you need to do is add a label to your template

`docker.versions.source=https://github.com/jlesage/docker-jdownloader-2`

**Comment by crovaxon** [comment]
Hello, thanks for the quick response!
I can confirm that the token gets set correctly, no whitespace of any kind. Do both kinds of tokens work (classic vs. fine-grained) or does it not matter what kind I generate?

But I see, so the 404 errors is likely not related to setting the token after all. I can confirm after removing the token that my currently two updateable docker containers are still exhibiting the 404 issue.
I am not just experiencing this with the jdownloader2 container, but also nextcloud:
```
HTTP error: 404. Response: {"message":"Not Found","documentation_url":"https://docs.github.com/rest/repos/repos#get-a-repository","status":"404"}
No releases found! ([Releases](https://api.github.com/repos/nextcloud/releases?per_page=100))
HTTP error: 404. Response: {"message":"Not Found","documentation_url":"https://docs.github.com/rest/repos/repos#get-a-repository","status":"404"}
No tags found! (https://api.github.com/repos/nextcloud/tags?per_page=100)
HTTP error: 404. Response: {"message":"Not Found","documentation_url":"https://docs.github.com/rest/repos/repos#get-a-repository","status":"404"}
No commits found! (https://api.github.com/repos/nextcloud/commits?per_page=100)
No releases found for 'https://github.com/nextcloud' !
https://api.github.com/repos/nextcloud/commits?per_page=100
org.opencontainers.image.source=https://github.com/nextcloud
org.opencontainers.image.version=
org.opencontainers.image.created=
docker.versions.source=
```

jdownloader updates very rarely so I am not sure if it has previously worked to fetch changelog there but with nextcloud my gutfeeling tells me that I was getting a changelog in earlier updates before or that at least it wasn't a 404 result. But obv. cannot confirm that right now.

That label I add to the template of the affected docker, right?

Will continue monitoring this, hopefully some of my other rather regularly updating dockers will get one soon, linuxserver based images usually always supplied a working changelog.

**Comment by crovaxon** [comment]
Just added the label to the jdownloader2 container and getting a changelog again:
![image](https://github.com/user-attachments/assets/5f05fb8a-9670-47d9-ab9d-d0c940e25a25)

**Comment by phyzical** [comment]
@crovaxon i would say the changelogs simply never worked.

It is outlined in the readme, but if a org.opencointainers.image label isnt on a image, then all we can do is fallback to the docker image name and do some optomistic guessing that the github repo will line up,

But even then sometimes the code is not managed in the same repo.

i.e sonarr and the linuxserver-sonarr.

So whenever you run into this its just a matter of looking for the secondary source

But for nextclouds case specifically i actually specify multiple as the autoguess just doesnt line up (image is called nextcloud/nextcloud but no github repo called that), and there is also two sources of changelogs, the container repo and the server

`docker.versions.source=https://github.com/nextcloud/docker,https://github.com/nextcloud/server` 

If you have any suggestions to help explain this better on the readme just LMK

**Comment by crovaxon** [comment]
I have waited for one of my other containers to get an update where I know that the changelog works as intended and these work fine. 👍 

I know not every image works to fetch the changelog just as you describe but I was convinced the output of those that do not support this plugin looked different, I did not remember seeing those 404's in the result. And after setting the token for myself and seeing the 404 results I mistakenly thought it was related, my bad. Sorry for the trouble.
